### PR TITLE
Obsolete methods that can be removed safely

### PR DESCRIPTION
**Pull request by fir3pho3nixx**
This addresses obsolescence from issue https://github.com/castleproject/Windsor/issues/338

This PR mainly targets:

 - ActAs
 - Parameters
 - ServiceOverrides 
 - Properties
 - Most methods in DefaultKernel_ObsoleteAPI
 - Most methods in WindsorContainer_ObsoleteAPI

Obsolete methods that can be removed safely:

 - Removed obsolete Resolve methods from IKernel/DefaultKernel && IWindsorContainer/WindsorContainer
 - Removed the obsolete AddComponent methods from IKernel/DefaultKernel
 - Removed the obsolete indexer from IKernel/DefaultKernel && IWindsorContainer/WindsorContainer
 - Removed the obsolete Properties methods from Component Registration
 - Removed the obsolete ServiceOverrides methods from Component Registration
 - Removed the obsolete Parameter method from Component Registration
 - Removed the obsolete ActAs method from Component Registrations and updated documentation

Obsolete methods that cannot be removed safely:

 - IKernel/DefaultKernel AddFacility(String key, IFacility facility)
 - IWindsorContainer/WindsorContainer AddFacility(String key, IFacility facility)

Still need to update CHANGELOG.md but will wait to avoid merge conflict.

### CODE REVIEW

File: src/Castle.Windsor/MicroKernel/DefaultKernel.cs [hunk 1]
```diff
@@ -278,9 +279,22 @@ IHandler IKernelInternal.CreateHandler(ComponentModel model)
...
+				throw new ArgumentException(
+					string.Format(
+						"Facility of type '{0}' has already been registered with the container. Only one facility of a given type can exist in the container.",
+						facilityType.FullName));
+			}
+			facilities.Add(facility);
+			facility.Init(this);
+
+			return this;
 		}
```
Comment (fir3pho3nixx) [inline src/Castle.Windsor/MicroKernel/DefaultKernel.cs]: After fixing this method to deprecate [the old one](https://github.com/castleproject/Windsor/pull/377/commits/161818cd63c2d33cd59d4090467f90d7a45d7663#diff-e1f78c645f8ca7366eb6563394edda3fL25), I came to the conclusion that IConfiguration as a parameter for the IFacility.Init method was no longer needed.

Comment (fir3pho3nixx) [reply src/Castle.Windsor/MicroKernel/DefaultKernel.cs]: Might have been a mistake. Not sure.

File: docs/fluent-registration-api-extensions.md [hunk 2]
```diff
@@ -65,26 +65,6 @@ container.Register(
 );
 ```
 
-### What does ActAs do?
```
Comment (jonorossi) [inline docs/fluent-registration-api-extensions.md]: Did you forget to remove `ActAs` from `ComponentRegistration`?

**Comment by jonorossi** [comment]
Looks good, just a comment about `ActAs`.

File: docs/fluent-registration-api-extensions.md (hunk 2 above)
Comment (fir3pho3nixx) [reply docs/fluent-registration-api-extensions.md]: Must have accidentally omitted that commit when I squished everything.

Comment (fir3pho3nixx) [reply docs/fluent-registration-api-extensions.md]: All fixed

**Comment by jonorossi** [comment]
All good.

**Comment by fir3pho3nixx** [comment]
Thanks. I will come back on the other PR tomorrow night.

**Pull request merged**